src/pages/Comments/hooks/usePostDetailPage.ts

```diff
@@ -0,0 +1,44 @@
+import { gql } from "graphql-request";
+import { useQuery } from "react-query";
+import { GetPostDetailPageQuery, GetPostDetailPageQueryVariables } from "src/generated/graphql";
+
+import useQueryClient from "src/hooks/useQueryClient";
+
+const usePostDetailPage = (postId: string) => {
+    const queryClient = useQueryClient();
+
+    const result = useQuery<GetPostDetailPageQuery>(
+        ["GetPostDetailPage"],
+        async () => {
+            const result = await queryClient.request<
+                GetPostDetailPageQuery,
+                GetPostDetailPageQueryVariables
+            >(
+                gql`
+                    query GetPostDetailPage($postId: uuid!) {
+                        posts(where: { postId: { _eq: $postId } }) {
+                            title
+                            postId
+                            content
+                            createdAt
+                            user {
+                                avatar
+                                name
+                            }
+                        }
+                    }
+                `,
+                { postId }
+            );
+
+            return result;
+        },
+        {
+            retry: false,

```

Comment:
Why not?

Reply:
Oh I want to show Error message at the first time so I changed it to false to show Error message. Since useSnakeBar has some problem, I will make hook option to default again.